guarddog/scanners/npm_package_scanner.py

```diff
@@ -18,35 +18,54 @@ def __init__(self) -> None:
         super().__init__(Analyzer(ECOSYSTEM.NPM))
 
     def download_and_get_package_info(self, directory: str, package_name: str, version=None) -> typing.Tuple[dict, str]:
-        git_target = None
-        if urlparse(package_name).hostname is not None and package_name.endswith('.git'):
-            git_target = package_name
+        git_package_name = self._parse_git_url(package_name)
 
-        if not package_name.startswith("@") and package_name.count("/") == 1:
-            git_target = f"https://github.com/{package_name}.git"
+        if git_package_name != "":
+            data, tarball_url = self._get_git_data_and_tarball_url(git_package_name, version)
+        else:
+            data, tarball_url = self._get_npm_data_and_tarball_url(package_name, version)
 
-        if git_target is not None:
-            raise Exception("Git targets are not yet supported for npm")
+        log.debug(f"Downloading NPM package from {tarball_url}")
+        file_extension = pathlib.Path(tarball_url).suffix
+        zippath = os.path.join(directory, package_name.replace("/", "-") + file_extension)
+        unzippedpath = zippath.removesuffix(file_extension)
+        self.download_compressed(tarball_url, zippath, unzippedpath)
+
+        return data, unzippedpath
+
+    def _parse_git_url(self, package_name: str) -> str:
+        parsed_url = urlparse(package_name)
+
+        path = parsed_url.path.strip("/")  # trim leading and trailing slashes
+
+        # TODO: support other git providers?
+        if parsed_url.hostname == "github.com" and \
+           path.endswith('.git') and \
+           not path.startswith("@") and \
+           path.count("/") == 1:
+            return path.removesuffix(".git")
+
+        return ""
 
+    def _get_git_data_and_tarball_url(self, package_name: str, version=None) -> typing.Tuple[dict, str]:
+        if version is None:
+            return {}, f"https://github.com/{package_name}/archive/refs/heads/main.zip"
```

Comment:
what if the main branch is `master`? (some old reposstill have this)

Reply:
Good point! I changed this URL in [a6853a5](https://github.com/DataDog/guarddog/pull/526/commits/a6853a543be0ba3899f471e465639a78b36ddd76) to use another endpoint which does not require to specify a branch and redirects to the default branch.

Reply:
Had to implement [cc257ec](https://github.com/DataDog/guarddog/pull/526/commits/cc257ecf5e1adab598a62f3138521b19ce470bcc) to make it work.